pkg/util/sync/pool.go

```diff
@@ -0,0 +1,42 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+// Package sync is utilities for synchronization
+package sync
+
+import "sync"
+
+// TypedPool is a type-safe version of sync.Pool
+type TypedPool[K any] struct {
```

Comment:
Why not just use `Pool[K any]` here?

Reply:
I don't feel super strongly, but on the off chance someone imports this package as `sync`, then using `sync.TypedPool` makes it less likely to be confused at-a-glance with the stdlib `sync.Pool`.